.github/workflows/instructlab-knowledge-e2e.yml

```diff
@@ -0,0 +1,58 @@
+# This workflow is designed to test the notebooks in "notebooks/instructlab-knowledge"
+# This job can be coppied for end to end tests of notebooks in other working directories
+
+name: Instructlab Knowledge End to End Tests
+
+# trigger on open pull requests only
+on:
+    pull_request:
+        branches:
+          - "main"
+        types: [opened, reopened]
+
+env:
+    # dependencies that need to be installed using pip for testing to work
+    TESTING_PIP_DEPENDENCIES: "papermill nbformat"
```

Comment:
Where is `nbformat` used here?

Reply:
its part of the validate notebook job we are awkwardly importing from instructlab right now. That will get cleaned up when I polish up and merge: https://github.com/instructlab/ci-actions/pull/15/files?short_path=155081c#diff-155081c742d9cc8de30b932236be2dbe0363c64fab0f2b2ccef0836012281fe9